src/guide/rust.md

```diff
@@ -187,63 +188,58 @@ Which is then **wrapped in a transaction** and **submitted to the peer** as [in
+Specifically, we have `AssetValueType::Quantity` which is effectively an unsigned 32-bit integer, a `BigQuantity` which is an unsigned 128 bit integer, which is enough to trade all possible IPV6 addresses, and quite possibly individual grains of sand on the surface of the earth and `Fixed`, which is a positive (though signed) 64-bit fixed-precision number with nine significant digits after the decimal point. It doesn't quite use binary-coded decimal for performance reasons. All three types can be registered as either **mintable** or **once-mintable**.
 
 Now, let's get back to coding. To register an asset, we first construct an `iroha_data_model::asset::DefinitionId` like so:
 
 ```rust
-use iroha_data_model::asset::DefinitionId;
-
-let id = DefinitionId {
-    name: "time".to_owned(),
-    domain_name: "looking_glass".to_owned(),
-};
+let id = AssetDefinitionId::from_str("time#looking_glass")?;
 ```
 
+::: info
+
+Note that we use `#` symbol to separate the name of the asset from the domain to which it belongs to. This is intentional. This reflects the rule that there can be many `alice`s in many domains, with only one `alice` per domain,  and there can be an asset that is also named `alice`, but there can be only one, regardless of type.
+
+:::
+
 Then construct an instruction
 
 ```rust
-let register_time = RegisterBox::new(IdentifiableBox::AssetDefinition(AssetDefinition::new(
-    id,
-    AssetValueType::Fixed,
-    false // If only we could mint more time.
-)));
+let register_time = RegisterBox::new(AssetDefinition::fixed(id).mintable_once().build());
+iroha_client.submit(register_time)?;
 ```
 
-For technical reasons, this would (_for now_) create a completely useless asset. In order to mint, one must already have an asset definition in the blockchain. A non-mintable asset definition, however cannot be minted by definition.
-
-This means that no matter how hard the _white_rabbit_ tries, the time that he has is the time that was given to him at genesis. And since we haven’t defined any time in the domain _looking_glass at_ genesis and defined time in a non-mintable fashion afterwards, the _white_rabbit_ is doomed to always be late.
+This creates an asset `time` that can only be minted once and has the type `fixed`. `AssetDefinition::fixed` just like its other cousins (`quantity` and `big_quantity`) returns a builder of an `AssetDefinition`. If you get odd error messages that's because you didn't call `build()` at the end.
```

Comment:
```suggestion
This creates an asset `time` that can only be minted once and has the type `fixed`. `AssetDefinition::fixed` just like its other cousins (`quantity` and `big_quantity`) returns a builder of an `AssetDefinition`. If you get odd error messages, that's because you didn't call `build()` at the end.
```